ngraph/core/reference/include/ngraph/runtime/reference/roi_align.hpp

```diff
@@ -0,0 +1,243 @@
+                                        sample_y = (T)sample_y_low;
+                                    }
+                                    else
+                                    {
+                                        sample_y_high = sample_y_low + 1;
+                                    }
+
+                                    if (sample_x_low >= feature_map_height - 1)
+                                    {
+                                        sample_x_high = sample_x_low = feature_map_width - 1;
+                                        sample_x = (T)sample_x_low;
+                                    }
+                                    else
+                                    {
+                                        sample_x_high = sample_x_low + 1;
+                                    }
+                                    pooling_points.push_back({sample_y_low, sample_x_low});
+                                    pooling_points.push_back({sample_y_low, sample_x_high});
+                                    pooling_points.push_back({sample_y_high, sample_x_low});
+                                    pooling_points.push_back({sample_y_high, sample_x_high});
+
+                                    // weight calculation for bilinear interpolation
+                                    auto ly = sample_y - static_cast<T>(sample_y_low);
+                                    auto lx = sample_x - static_cast<T>(sample_x_low);
+                                    auto hy = static_cast<T>(1.) - ly;
+                                    auto hx = static_cast<T>(1.) - lx;
+
+                                    pooling_weights.push_back(hy * hx);
+                                    pooling_weights.push_back(hy * lx);
+                                    pooling_weights.push_back(ly * hx);
+                                    pooling_weights.push_back(ly * lx);
+                                }
+                            }
+                        }
+                    }
+
+                    std::vector<T> tmp_out;
+
+                    for (uint64_t channel_index = 0; channel_index < C; channel_index++)
+                    {
+                        tmp_out.reserve(pooled_height * pooled_width);
+                        uint64_t sample_index = 0;
+                        for (int64_t y_bin_ind = 0; y_bin_ind < pooled_height; y_bin_ind++)
+                        {
+                            for (int64_t x_bin_ind = 0; x_bin_ind < pooled_width; x_bin_ind++)
+                            {
+                                T pooled_value = 0;
+                                for (int64_t bin_sample_ind = 0;
+                                     bin_sample_ind < num_samples_in_bin;
+                                     bin_sample_ind++)
+                                {
+                                    auto s1 = feature_maps[feature_maps_transform.index(
```

Comment:
I think the s1..s4 variables should be changed to something more elaborate.

Reply:
Changed the variable names, also added a comment above to maybe clarify what those variables are used for.